test/common/router/config_impl_test.cc

```diff
@@ -5780,6 +5787,95 @@ name: foo
   }
 }
 
+TEST_F(RouteMatcherTest, TlsContextMatching) {
```

Comment:
Please add a brief comment above this test explaining what it does.

Reply:
I've added a comment - let me know if it needs more details. thanks